R/content.R

```diff
@@ -273,6 +273,56 @@ Content <- R6::R6Class(
       cat('content_item(client, guid = "', self$get_content()$guid, '")', "\n", sep = "")
       cat("\n")
       invisible(self)
+    },
+
+    #' @description Render or restart this content item.
+    refresh = function() {
+      if (is_rendered(self$content$app_mode)) {
+        self$render()
+      } else if (is_interactive(self$content$app_mode)) {
+        self$restart()
+      } else {
+        warning("Content app mode cannot be rendered or restarted.")
+      }
+    },
+
+    #' @description Render this content item. If parameterized, the default
+    #' variant is rendered by default. Only operates on rendered content types.
+    render = function() {
+      # TODO: better messages
+      if (!is_rendered(self$content$app_mode)) {
+        warning("This content cannot be rendered because it is not a rendered content type.")
+        return(invisible(self))
+      }
+      rendered <- self$default_variant$render()
+      rendered$task_id <- rendered$id
+    
+      # TODO: Wait and return invisible(self)?
+      # VariantTask$new(connect = self$default_variant$get_connect(), content = self$get_content(), key = self$default_variant$key, task = rendered)
+      invisible(self)
+    },
+
+    #' @description Restart this application. Only operates on interactive content.
+    restart = function() {
+      if (!is_interactive(self$content$app_mode)) {
+        warning("This content cannot be restarted because it is not an interactive content type.")
+        return(invisible(self))
+      }
+      random_hash = token_hex(32)
+      # https://rlang.r-lib.org/reference/glue-operators.html#what-s-the-deal-with-
+      self$environment_set("{random_hash}" = random_hash)
+      self$environment_set("{random_hash}" = NA)
```

Comment:
Is the link to the glue documentation the right anchor? it looks like it's the `:=` operator?

Reply:
Yeah, the right anchor is `https://rlang.r-lib.org/reference/glue-operators.html#using-glue-syntax-in-packages`

```suggestion
      # https://rlang.r-lib.org/reference/glue-operators.html#using-glue-syntax-in-packages
      self$environment_set("{random_hash}" = random_hash)
      self$environment_set("{random_hash}" = NA)
```